2_run/src/run_glm3.R

```diff
@@ -0,0 +1,150 @@
+    mutate(date = as.Date(lubridate::floor_date(DateTime, 'days'))) %>% select(-DateTime)
+  glmtools::get_var(nc_filepath, var_name = 'hice') %>%
+    dplyr::mutate(ice = hice > 0, date = as.Date(lubridate::ceiling_date(DateTime, ' days'))) %>% dplyr::select(-hice, -DateTime) %>%
+    dplyr::left_join(temp_data, ., by = 'date') %>%
+    select(time = date, everything()) %>%
+    feather::write_feather(export_fl)
+}
+
+#' @title run glm3 simulation
+#' @description For each lake - gcm - time period combo, write the nml
+#'  file, the meteo data, and run the glm3 model
+#'  @param sim_dir base directory for simulations
+#'  @param nml_objs list of nml objects (one per lake id), named by lake id
+#'  @param meteo_xwalk a lake-gcm-time_period crosswalk table with the 
+#'  meteo file name and hash for each model run
+run_glm3_model <- function(sim_dir, nml_objs, meteo_xwalk, export_fl_template) {
+  # pull lake_id from meteo_xwalk
+  lake_id <- meteo_xwalk$site_id
+  time_period <- meteo_xwalk$time_period
+  gcm <- meteo_xwalk$gcm
+  
+  # prepare to write inputs and results locally for quick I/O
+  sim_lake_dir <- file.path(sim_dir, sprintf('%s_%s_%s', lake_id, gcm, time_period))
+  dir.create(sim_lake_dir, recursive=TRUE, showWarnings=FALSE)
+  # delete sim_lake_dir after model has run and we've extracted the results
+  on.exit(unlink(sim_lake_dir, recursive = TRUE))
+  
+  # Define time period begin and end dates
+  times <- strsplit(time_period,'_')[[1]]
+  time_period_begin <- sprintf('%s-01-01', times[1])
+  time_period_end <- sprintf('%s-12-31', times[2])
+  
+  # Read in meteo data, add burn in and burn out and save to sim_lake_dir
+  meteo_data <- munge_meteo_w_burn_in_out(meteo_xwalk, time_period_begin, time_period_end, 
+                                          burn_in = 300, burn_out = 190)
+  sim_meteo_filename <- 'meteo_fl.csv'
+  readr::write_csv(meteo_data, file.path(sim_lake_dir, sim_meteo_filename))
+  
+  # Define simulation start and stop dates using munged meteo data
+  sim_start <- format(meteo_data[1,]$time, "%Y-%m-%d")
+  sim_stop <- format(tail(meteo_data, 1L)$time, "%Y-%m-%d")
+  
+  # write nml file, specifying meteo file and start and stop dates:
+  nml_obj <- nml_objs[[lake_id]]
+  nml_obj <- set_nml(nml_obj, arg_list = list(nsave = 24, 
+                                              meteo_fl = sim_meteo_filename,
+                                              sim_name = sprintf('%s_%s_%s', lake_id, gcm, time_period),
+                                              start = sim_start,
+                                              stop = sim_stop))
+  nml_obj$light$Kw_file <- NULL # use static value instead of relying on a file
+  glmtools::write_nml(nml_obj, file.path(sim_lake_dir, 'glm3.nml'))
+  
+  # Set up export tibble
+  export_fl <- sprintf(export_fl_template, lake_id, gcm, time_period)
+  export_tibble <- tibble(
+    run_date = Sys.time(),
```

Comment:
I know you and I talked about this but after reviewing this and thinking about the linkages of targets more, I am questioning whether we want to save `run_date`. I am nervous about this scenario: we change something in the model code, re-run it, and get the same results. So the `export_hash` is the same as it was before; however, the `run_date` is now different. This means that one of the columns in `p2_glm_uncalibrated_run_groups` is now different from before and would trigger a build of `p2_glm_uncalibrated_output_feathers` for that group. Having the `run_date` is great for diagnostics, though. I wonder if we could edit `p2_glm_uncalibrated_run_groups` so that right before the `group_by()`, we do a `select(-run_date)` with a comment about how it is used for diagnostic purposes but the `export_hash` is the authority on whether the downstream steps should be run. Does that make sense?

Reply:
yes, absolutely. That's a good catch. I think the same could be true of the `glm_time_s` column b/c that doesn't seem to be consistent over time, even if nothing else has changed. I wonder if we should just reduce the tibble to `lake_id`, `gcm`, `time_period`, `export_fl` and `export_hash` before the `group_by()`. That way we can keep all the glm-specific diagnostics in the original tibble output by building `p2_uncalibrated_glm_runs`, but only rely on the `export_hash` column as the indicator of whether the downstream steps should be run.

Reply:
I think this makes sense - same idea with the `select(-run_date)` above. I think you are right, we don't want `glm_time_s` triggering rebuilds. I think `select(lake_id, gcm, `time_period, export_fl, export_hash)`, as you suggested, would be the right approach.

Reply:
Okay - I added in this approach, as well as a fix to not track previously created files, since I noticed that bug while I was back in the code. I tested out the fix that we chatted about a few times, and it works as expected.